### DISCUSSION TITLE
Intercept on request function randomly with every test run

### DISCUSSION THREAD

**Discussion by TesterUnknownMAir** [cypress-io/cypress#15781] [Questions and Help]
Hi
I'm trying to write wait on some requests & this is my code:
`        cy
            .intercept('POST', '**/api/Availability')
            .as('availabilecheck');
        cy
            .wait('@availabilecheck')
            .its('response.statusCode')
            .should('eq', 200)`
The problem is that the wait runs on a different request each time I run the test.
Each time it selects a different request to wait on, completely randomly.
What is the problem exactly?

**Comment by lmiller1990** [comment]
Can you provide a minimal reproduction? Based on the snippet above, everything is fine (your code is the same as [what we have in our docs](https://docs.cypress.io/api/commands/intercept#Matching-URL).
